src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h

```diff
@@ -0,0 +1,148 @@
+ * -----------------------------------------------------------------------
+ * |  int32 length  | int32 requestID | int32 responseTo|  int32 opCode  |
+ * -----------------------------------------------------------------------
+ *
+ * Documentation regarding OP_MSG frames can be found here:
+ * https://github.com/mongodb/specifications/blob/master/source/message/OP_MSG.rst#sections
+ *
+ * A frame of type OP_MSG looks like:
+ * -----------------------------------------------------------------------
+ * |  int32 length  | int32 requestID | int32 responseTo|  int32 opCode  |
+ * -----------------------------------------------------------------------
+ * | uint32 flagBits|  Sections[] sections  |  optional uint32 checksum  |
+ * -----------------------------------------------------------------------
+ *
+ * There must be only one payload section of kind 0 and any number of payload sections of kind 1.
+ *
+ * A payload section of kind 0 looks like:
+ * -----------------------------------------------------------------------
+ * |  uint8 kind | int32 document payload length |        document       |
+ * -----------------------------------------------------------------------
+ *
+ * A payload section of kind 1 looks like:
+ * -----------------------------------------------------------------------
+ * |  uint8 kind  |               int32 section length                   |
+ * -----------------------------------------------------------------------
+ * |       identifier (command argument that ends with \x00)             |
+ * -----------------------------------------------------------------------
+ * | int32 document payload length |              document               |
+ * -----------------------------------------------------------------------
+ *
+ * There can be 0 or more documents in a section of kind 1 without a separator between them.
+ */
+
+struct Frame : public FrameBase {
+  // Message Header Fields
+  // Length of the mongodb header and the wire protocol data.
+  int32_t length = 0;
+  int32_t request_id = 0;
+  int32_t response_to = 0;
+  int32_t op_code = 0;
+
+  // OP_MSG Fields
+  // Relevant flag bits
+  bool checksum_present = false;
+  bool more_to_come = false;
+  bool exhaust_allowed = false;
+  std::vector<Section> sections;
+  std::string op_msg_type;
+  uint32_t checksum = 0;
+
+  bool consumed = false;
+  size_t ByteSize() const override { return sizeof(Frame); }
+
+  std::string ToString() const override {
+    return absl::Substitute(
+        "MongoDB message [length=$0 requestID=$1 responseTo=$2 opCode=$3 checksum=$4]", length,
+        request_id, response_to, op_code, checksum);
+  }
+};
+
+struct Record {
+  Frame req;
+  Frame resp;
+
+  std::string ToString() const {
+    return absl::Substitute("req=[$0] resp=[$1]", req.ToString(), resp.ToString());
+  }
+};
+
+struct ProtocolTraits : public BaseProtocolTraits<Record> {
+  using frame_type = Frame;
+  using record_type = Record;
+  using state_type = NoState;
+};
+
```

Comment:
I think we should add the `src/stirling/source_connectors/socket_tracer/protocols/mongodb/BUILD.bazel` file changes to use this source file within this PR (cc_library / pl_cc_library).